x/axelarcork/client/cli/tx.go

```diff
@@ -0,0 +1,491 @@
+			if err != nil {
+				return err
+			}
+
+			proposal := types.AddAxelarManagedCellarIDsProposalWithDeposit{}
+			contents, err := os.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			if err = clientCtx.Codec.UnmarshalJSON(contents, &proposal); err != nil {
+				return err
+			}
+
+			deposit, err := sdk.ParseCoinsNormalized(proposal.Deposit)
+			if err != nil {
+				return err
+			}
+
+			for _, id := range proposal.CellarIds.Ids {
+				if !common.IsHexAddress(id) {
+					return fmt.Errorf("%s is not a valid ethereum address", id)
+				}
+			}
+
+			content := types.NewAddManagedCellarIDsProposal(
+				proposal.Title,
+				proposal.Description,
+				"",
+				0,
+				&types.CellarIDSet{Ids: proposal.CellarIds.Ids})
+
+			name, chainID, err := GetChainInfoFromFlags(cmd)
+			if err != nil {
+				return err
+			}
+			content.ChainName = name
+			content.ChainId = chainID
+
+			from := clientCtx.GetFromAddress()
+			msg, err := govtypes.NewMsgSubmitProposal(content, deposit, from)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	AddChainFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdSubmitRemoveProposal implements the command to submit a cellar id removal proposal
+func GetCmdSubmitRemoveProposal() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remove-cellar-id [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a cellar id removal proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a cellar removal proposal along with an initial deposit.
+The proposal details must be supplied via a JSON file.
+
+Example:
+$ %s tx gov submit-proposal remove-cellar-id <path/to/proposal.json> --from=<key_or_address>
+
+Where proposal.json contains:
+
+{
+  "title": "Dollary-doos LP Cellar Removal Proposal",
+  "description": "I don't trust them",
+  "cellar_ids": ["0x123801a7D398351b8bE11C439e05C5B3259aeC9B", "0x456801a7D398351b8bE11C439e05C5B3259aeC9B"],
+  "deposit": "10000usomm"
+}
+`,
+				version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			proposal := types.RemoveAxelarManagedCellarIDsProposalWithDeposit{}
+			contents, err := os.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			if err = clientCtx.Codec.UnmarshalJSON(contents, &proposal); err != nil {
+				return err
+			}
+
+			deposit, err := sdk.ParseCoinsNormalized(proposal.Deposit)
+			if err != nil {
+				return err
+			}
+
+			for _, id := range proposal.CellarIds.Ids {
+				if !common.IsHexAddress(id) {
+					return fmt.Errorf("%s is not a valid ethereum address", id)
+				}
+			}
+
+			content := types.NewRemoveManagedCellarIDsProposal(
+				proposal.Title,
+				proposal.Description,
+				"",
+				0,
+				&types.CellarIDSet{Ids: proposal.CellarIds.Ids})
+
+			name, chainID, err := GetChainInfoFromFlags(cmd)
```

Comment:
Bumping this. Especially given that we're only using chain ID now, I think rather than having an additional flag and reading it we should just have it be a required argument, it is necessary for all the axelar commands. And as mentioned above, I am worried about how this might interact with the dynamically constructed gov client.